app/models/manageiq/providers/ibm_cloud/vpc/manager_mixin.rb

```diff
@@ -16,7 +16,12 @@ def connect(options = {})
     key = authentication_key(options[:auth_type])
     region = options[:provider_region] || provider_region
     sdk = self.class.raw_connect(key)
-    sdk.vpc(:region => region)
+    if options[:service] == 'events'
+      service_key = authentication_key("events")
+      sdk.events(:region => region, :service_key => service_key)
+    else
+      sdk.vpc(:region => region)
+    end
```

Comment:
Not necessary for this PR, but this feels like it should be broken up into smaller methods and/or the raw_connect method should take the "service" as a parameter (I think that's how we do it for amazon).

@agrare Thoughts on this?  I know in other places you have individual methods like `def vpc`, `def events`, but I can't recall at what level you had those methods.

Reply:
Also with this addition the comment on line 12 and 14 is now Basmati Rice (https://twitter.com/nzkoz/status/538892801941848064)

Reply:
Passing the service in as an option and switching on it is pretty common, you're thinking of the google cloud collector where I was caching the various service connections.

We could bump this logic up to the `.raw_connect` method if that would help but at this point that is only used by this method and `verify_credentials` which doesn't need any specific service (just checking the IAM token is valid)